single-container/Dockerfile

```diff
@@ -83,9 +82,13 @@ RUN tar xf catalog_qt_2.tar
 # and run it
 # The reason for keeping this file outside of Dockerfile is that file itself is quite
 # large and, additionally, it depends on source layout.
-COPY --chown=imas scripts/compile-java-code.sh /home/imas/opt/compile-java-code.sh
-RUN chmod +x /home/imas/opt/compile-java-code.sh
-RUN /home/imas/opt/compile-java-code.sh
+COPY --chown=imas scripts/compile-scripts/* /home/imas/opt/
+RUN chmod +x /home/imas/opt/compile-*.sh
+
+RUN /home/imas/opt/compile-common.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-plugin.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-client.sh /home/imas/opt/catalog_qt_2
+RUN /home/imas/opt/compile-server.sh /home/imas/opt/catalog_qt_2
```

Comment:
Usually, subsequent `RUN` lines are combined with the `&&` operator. This makes the resulting image smaller.

Reply:
OK - I will change that to

```
RUN /home/imas/opt/compile-common.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-plugin.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-client.sh /home/imas/opt/catalog_qt_2 && \
/home/imas/opt/compile-server.sh /home/imas/opt/catalog_qt_2
```

Reply:
> IMHO, it would be beneficial to use a staged building process in Dockerfile. One stage will build the Java projects. Others will just copy the resulting binaries (i.e. JARs). The rest is left to Docker caching mechanisms. This way, the Java project is compiled only once per introduced changes. Maybe in the future, we can think of a solution like that.

Yes, exactly. I think we definitely should move towards this approach, but I have to make some research here. The major issue here is not the build process itself - it takes just a minute or so. The biggest problem is related to `maven` downloading all the dependencies each time we run `build` command.

There is a way to overcome this issue. We can extract only `pom.xml` files, download dependencies in one of the early stages, and then build on top of that. This way, we will avoid downloading dependencies as long as `pom.xml` files stay the same. However, there is one issue. We have to change the way of building a little bit.

Now, `catalog_qt_2.tar` is loaded into image as one of stages. So, we are already lost because it will trigger all the subsequent stages. This is why instead of passing `tar` file I have to pass directory, and before copying directory into image, I have to copy just a `pom.xml` files in a first place. This is doable but requires little bit of the effort to prepare and test the whole solution.